indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskMaster.java

```diff
@@ -203,6 +203,9 @@ public void stop()
     giant.lock();
 
     try {
+      if (isLeader()) {
+        leadershipListener.stopBeingLeader();
```

Comment:
Is this racy at all? can I loose leadership between `isLeader` and here, then exception out and never unregister my listener?

Reply:
Hmm, I'm not sure... should this go after unregister or be allowed to silently fail?

Reply:
Hmm, upon further looking at this I _think_ if lifecycle stop handlers on the leader lifecycle exploded while stopping it would indeed break out and prevent the listener unregister and probably wreck the outer lifecycle module's lifecycle stop too. I'll wrap this bit in a try/catch to fail silently since we are stopping anyway, and would cover this situation.